packages/teuchos/cmake/Dependencies.cmake

```diff
@@ -9,3 +9,5 @@ TRIBITS_PACKAGE_DEFINE_DEPENDENCIES(
     KokkosCompat  kokkoscompat  PS  OPTIONAL
     KokkosComm    kokkoscomm    PS  OPTIONAL
   )
+
+SET(LIB_OPTIONAL_DEP_TPLS Valgrind)
```

Comment:
This does not do what you think it does.  This dependency needs to be moved to one of the Teuchos subpackages, likely:

* teuchos/core/cmake/Dependencies.cmake

Reply:
Yup, because of the define `HAVE_TEUCHOS_TIME_MASSIF_SNAPSHOTS` added to teuchos/core/cmake/Teuchos_config.h.in, you have to move the optional TPL enalbe to teuchos/core/cmake/Dependencies.cmake.

Reply:
I don't understand how this is working as is.  The include path for valgrind.h must be getting added in some other way.

Reply:
I moved the dependency to teuchos/core/cmake/Dependencies.cmake .
It also compiles and works on a system with Valgrind in a non-standard location.
Is this the correct way of including the TPL?